qutip/solver/result.py

```diff
@@ -916,10 +926,12 @@ def __add__(self, other):
             new.expect = new.average_expect
             new.e_data = new.average_e_data
 
-            new.std_e_data = {
-                k: np.sqrt(avg_expect2 - abs(avg_expect**2))
-                for k, avg_expect, avg_expect2 in zip(self._raw_ops, avg, avg2)
-            }
+            new.std_e_data = {}
+            for i, key in enumerate(self._raw_ops):
+                std2 = avg2[i] - abs(avg[i]**2)
+                std2[std2 < 0] = 0.
```

Comment:
Floating point calculations are not exact, values that should be 0 can very small negative numbers and raise a warning for the square root. This is not an issue the user can deal with, it will just happen sometime when subtracting large floating point number.
For example, when I compute the std of 1000 measurement of 4/3 I get:
```
>>> N = 1000
>>> v = 4/3
>>> x = [v]*N
>>> x2 = [v**2]*N
>>> (sum(x)/N)**2 - sum(x2) / N
-5.040412531798211e-14
```

Reply:
Thanks for the detailed explanation!